File: Java/bubble_sort.java
import java.io.*;
class bubble_sort
{
    public static void main(String args[])throws IOException
    {
        BufferedReader buf=new BufferedReader(new InputStreamReader(System.in));
        int N;
        System.out.println("Enter the number of elements");
        N=Integer.parseInt(buf.readLine());
        int arr[]=new int[N];
        System.out.println("Enter the elements of array");
        for(int i=0;i<N;i++){
            arr[i]=Integer.parseInt(buf.readLine());
        }
        System.out.println("Original Array");
       for(int i=0;i<N;i++){
       System.out.print(arr[i]+"  ");
       }
       System.out.println();
       int temp;
       for(int i=0;i<N-1;i++){
           for(int j=0;j<N-i-1;j++){
              if(arr[j]>arr[j+1]){
              temp=arr[j];
              arr[j]=arr[j+1];
              arr[j+1]=temp;
              }
           }
       }
System.out.println("Sorted Array");
for(int i=0;i<N;i++){
  System.out.print(arr[i]+"  ");
}
System.out.println();
}}

File: C/bubble_sort.c
#include <stdio.h>
#include <stdlib.h>

int main(){
int N; //Number of elements in the array
printf("\n Enter the number of elements in the array\n");
scanf("%d",&N);
int *arr=malloc(N*sizeof(*arr)); //Dynamic creation of an Integer array of Size N
printf("\n Enter the elemnts of the array\n");
for(int i=0;i<N;i++){
  scanf("%d",&arr[i]); //Inputting the elements of array
}
printf("\n Original Array\n"); //Printing the original array
for(int i=0;i<N;i++){
  printf("%d ",arr[i]);
}
int temp; //Temporary variable used for swapping later
for(int i=0;i<N-1;i++){ //Sorting process begins
  for(int j=0;j<N-i-1;j++){
    if(arr[j]>arr[j+1]){ //Checking if a number at the previous index is greater than the number at its next,if so, swapping takes place
      temp=arr[j];
      arr[j]=arr[j+1];
      arr[j+1]=temp;
    }
  }
}
printf("\n Sorted Array\n"); //Printing sorted array
for(int i=0;i<N;i++){
  printf("%d ",arr[i]);
}
}
